crates/metadata-server/src/raft/server.rs

```diff
@@ -429,8 +429,20 @@ impl RaftMetadataServer {
     async fn load_initial_state_from_local_metadata_server(
         &mut self,
     ) -> anyhow::Result<KvMemoryStorage> {
+        let local_metadata_server_options = MetadataServerOptions::default();
+
+        // check whether local metadata server data exists to avoid wrong intentions and possible
+        // misconfigurations
+        if !local::storage::RocksDbStorage::data_dir_exists() {
+            anyhow::bail!(
+                "Trying to migrate from local metadata server but couldn't find any data. \
+            Please make sure that this node has been run with the local metadata server before or \
+            unset the metadata-server.migrate-from-local-metadata-server option."
+            );
```

Comment:
If migration is automatic, then we need to be a little more graceful here. We just need to check if the directory exists or not to decide on opening the database, but if it doesn't, we can move on without migration.

Reply:
Yes agreed. Let's discuss it here: https://github.com/restatedev/restate/pull/2708#discussion_r1952415984.